### ISSUE TITLE
[Feature Request] Back to "clean" homescreen

### ISSUE THREAD

**Issue by 4r1-60ld** [MM2-0/Kvaesitso#291]
Hello,

Having trouble discribing what i mean, but i have noticed that when you for an instance open an app or use the widget screen and then press back, you get back to where you "left off" i.e. the widget panel is open or the app drawer. Could we have an option to go back to a "clean" homescreen like when the phone has booted?

**Comment by CDTay** [comment]
This feels essential and I'm honestly confused why it's not already a feature

**Comment by MM2-0** [comment]
App drawer should close if you return to the home screen after more than a minute (sometimes it is beneficial to return to where you were, e.g. when you accidentially clicked the wrong search result).

I should probably apply the same logic to the widgets.

**Comment by DasDunkel** [comment]
> App drawer should close if you return to the home screen after more than a minute (sometimes it is beneficial to return to where you were, e.g. when you accidentially clicked the wrong search result).
> 
> I should probably apply the same logic to the widgets.
@MM2-0 
A good compromise for this would be to add an option to change the time it waits, I do like being able to go back to my search if I opened the wrong app, but 1 minute is a bit too excessive for this and (for me) leads to unnecessary scrolling, being able to set it to something like 5 seconds would be a major improvement

**Comment by mammuth** [comment]
> App drawer should close if you return to the home screen **after more than a minute**

I and a lot of other people (https://github.com/MM2-0/Kvaesitso/issues/497, https://github.com/MM2-0/Kvaesitso/issues/1478, https://github.com/MM2-0/Kvaesitso/issues/1106) didn't figure this out and either thought the current behavior is a bug or a missing feature.

>  (sometimes it is beneficial to return to where you were, e.g. when you accidentially clicked the wrong search result).

This reasoning seems fair. 60s sounds very extreme, though (and personally, I'd never want to have the current behavior, I think. If I navigate to _home_ (and not _back_), I want to get to _home_).

@MM2-0 What's your take on either **turning those 60s down (closer to eg. 5 or 10s), or alternatively make it configurable**? 🤔

**Comment by MM2-0** [comment]
> @MM2-0 What's your take on either turning those 60s down (closer to eg. 5 or 10s), or alternatively make it configurable? 🤔

I already did that. Currently, the launcher resets itself after 5 seconds of inactivity.

**Comment by CaviaPorcellus** [comment]
I get going back to the app drawer/search screen when hitting back, but what's the reasoning being home doing the same thing? Almost filed another duplicate bug because I assumed the behavior with regards to the home button was one.

**Comment by nvourva** [comment]
> > @MM2-0 What's your take on either turning those 60s down (closer to eg. 5 or 10s), or alternatively make it configurable? 🤔
> 
> I already did that. Currently, the launcher resets itself after 5 seconds of inactivity.

Has this feature been released? I'm on 1.39.3-fdroid on Android 16 and the search bar never resets. Whenever I go back, I end up in the search bar with the bar focused and whatever I typed in there.

**Comment by MM2-0** [comment]
> Has this feature been released?

Yes, ~3 years ago

**Comment by nvourva** [comment]
In that case it seems that the timeout does not work, at least for my two devices.

**Comment by MM2-0** [comment]
Yes, that's why it's labeled as bug and can't reproduce. But I can't reproduce it, so I don't know how to fix this.

**Comment by nvourva** [comment]
I noticed that this is consistently written in the debug logs whenever I reproduce it. Could it be connected? Does it make sense to help the entire log file?

```
02-23 18:24:19.396 WindowOnBackDispatcher
sendCancellfRunning: isInProgress=false
callback=androidx.navigationevent
.OnBackInvokedInput$createOnBackAnimationCallback$1@34cfe73
```

**Comment by MM2-0** [comment]
I don't think so. What devices / Android version do you all use?

**Comment by nvourva** [comment]
Ok, I think I might have found something. I can consistently make this behavior appear/disappear by setting the swipe down gesture to open quick settings. If I set it to open the quick settings the search bar does not go away when I back out of an app even after 30+ seconds, if I set it to open search (as is by default) backing out of an app after 10secs goes to the homescreen.

Tested on Motorola Edge 60 Pro - Android 16 and Motorola Edge 30 Neo - Android 14

**Comment by MM2-0** [comment]
Oh, now I can reproduce it too. Thank you! When you don't have any swipe gesture assigned to search, it won't close.

**Comment by nvourva** [comment]
Awesome! Happy to help!
